lib/semantics/resolve-labels.cc

```diff
@@ -217,12 +216,8 @@ struct UnitAnalysis {
 
 class ParseTreeAnalyzer {
 public:
-  ParseTreeAnalyzer() {}
-  ParseTreeAnalyzer(ParseTreeAnalyzer &&that)
-    : programUnits_{std::move(that.programUnits_)},
-      errorHandler_{std::move(that.errorHandler_)}, currentPosition_{std::move(
-                                                        that.currentPosition_)},
-      constructNames_{std::move(that.constructNames_)} {}
+  ParseTreeAnalyzer(parser::Messages &errorHandler)
```

Comment:
I think it's the same as the implicitly-declared move constructor so it wouldn't change anything to take it out.
We could also leave it in with "= default".

Reply:
+1 for `=default`.

Reply:
I don't recommend defining a move constructor *just* to disable a copy constructor, of course.  When going out of one's way to preclude copy construction (as I did in the parse tree data structures so that I would never have to worry about finding expensive accidental deep copies that should have been moves), there's `=delete`.  For data structures that constitute a more general-purpose library with unknown clients, it's best to support both copy and move semantics for construction and assignment.  I really prefer `=default` for these instead of explicit constructors so that we don't have to worry about missing the construction of a member when adding new ones to a class.  Another technique that is good is to support only move construction and assignment in the class, and supply a `Clone()` member function to create explicit copies for the rare cases when a copy is really needed (and to make them stand out).